blockchain/validate.go

```diff
@@ -757,283 +739,86 @@ func checkBlockSanityContextFree(block *dcrutil.Block, timeSource MedianTimeSour
-				"votes: %v)", headerApproves, votesApprove)
-			return ruleError(ErrIncongruentVotebit, errStr)
-		}
-	}
-
-	// A block must not contain anything other than ticket purchases or
-	// treasury transactions (when agenda is enabled) prior to stake
-	// validation height.
-	if header.Height < stakeValidationHeight {
-		if int64(len(msgBlock.STransactions)) !=
-			totalTickets+totalTAdd+totalTreasurybase {
-			errStr := fmt.Sprintf("block contains disallowed stake"+
-				" transactions before stake validation height "+
-				"%d (total: %d, expected %d)",
-				uint32(chainParams.StakeValidationHeight),
-				len(msgBlock.STransactions), header.FreshStake)
-			return ruleError(ErrInvalidEarlyStakeTx, errStr)
-		}
+		str := fmt.Sprintf("block header commitment to %d revocations does "+
+			"not match %d contained in the block", header.Revocations,
+			totalRevocations)
+		return ruleError(ErrRevocationsMismatch, str)
 	}
 
-	// Check for duplicate transactions.  This check will be fairly quick
-	// since the transaction hashes are already cached due to building the
-	// merkle trees above.
+	// Check for duplicate transactions.
 	existingTxHashes := make(map[chainhash.Hash]struct{})
 	stakeTransactions := block.STransactions()
 	allTransactions := append(transactions, stakeTransactions...)
-
 	for _, tx := range allTransactions {
 		hash := tx.Hash()
 		if _, exists := existingTxHashes[*hash]; exists {
-			str := fmt.Sprintf("block contains duplicate "+
-				"transaction %v", hash)
+			str := fmt.Sprintf("block contains duplicate transaction %v", hash)
 			return ruleError(ErrDuplicateTx, str)
 		}
 		existingTxHashes[*hash] = struct{}{}
 	}
 
-	// The number of signature operations must be less than the maximum
-	// allowed per block.
-	totalSigOps := 0
-	for _, tx := range allTransactions {
-		// We could potentially overflow the accumulator so check for
-		// overflow.
-		lastSigOps := totalSigOps
-
-		msgTx := tx.MsgTx()
-		isCoinBase := standalone.IsCoinBaseTx(msgTx, isTreasuryEnabled)
-		isSSGen := stake.IsSSGen(msgTx, isTreasuryEnabled)
-		totalSigOps += CountSigOps(tx, isCoinBase, isSSGen,
-			isTreasuryEnabled)
-		if totalSigOps < lastSigOps || totalSigOps > MaxSigOpsPerBlock {
-			str := fmt.Sprintf("block contains too many signature "+
-				"operations - got %v, max %v", totalSigOps,
-				MaxSigOpsPerBlock)
-			return ruleError(ErrTooManySigOps, str)
-		}
-	}
-
 	return nil
 }
 
 // CheckBlockSanity performs some preliminary checks on a block to ensure it is
 // sane before continuing with block processing.  These checks are context
 // free.
-func CheckBlockSanity(block *dcrutil.Block, timeSource MedianTimeSource, chainParams *chaincfg.Params, isTreasuryEnabled bool) error {
-	return checkBlockSanity(block, timeSource, BFNone, chainParams, isTreasuryEnabled)
+func CheckBlockSanity(block *dcrutil.Block, timeSource MedianTimeSource, chainParams *chaincfg.Params) error {
+	return checkBlockSanity(block, timeSource, BFNone, chainParams)
 }
```

Comment:
One other comment - wouldn't this technically require a version bump since it's changing an exported function signature?

Reply:
blockchain was bumped to v4 in bad849b69df07ab5b98bfeb871dfef6d6fa2bf44 and hasn't been tagged with a stable version yet, so breaking changes are permitted right now.

Reply:
Makes sense, thanks for the clarification.